### DISCUSSION TITLE
How to use optimistic lock version in cascade entity

### DISCUSSION THREAD

**Discussion by cqzhuhui** [micronaut-projects/micronaut-data#1960] [Q&A]
I already update micronaut version to latest and micronaut-data is 3.9.1. I don't know whether it is a bug of micronaut-data.
in my entity, there is a field use annotation below:
```
data class Entity1(
@Relation(Relation.Kind.MANY_TO_ONE, cascade = [Cascade.ALL])
val verification: Entity2,
@field:Version val version: Long? = 0L
)
```
when I use repository.update(entity1), I found the generated sql without version like below:
`10:10:34.134 [Test worker] DEBUG io.micronaut.data.query - Executing SQL query: UPDATE "entity2" SET "deleted_at"=?,"updated_at"=? WHERE ("id" = ?)`
And the updated sql for entity1 is correct like below:
`10:10:34.139 [Test worker] DEBUG io.micronaut.data.query - Executing SQL query: UPDATE "entity1" SET "entity1_id"=?,"version"=? WHERE ("id" = ? AND "version" = ?)`

**Comment by cqzhuhui** [comment]
I want to check the optimistic lock of entity2 when I update entity1. And the entity2 is defined like below:
data class Entity2(
    @DateCreated(truncatedTo = ChronoUnit.MILLIS) val createdAt: Instant? = null,
    @DateUpdated(truncatedTo = ChronoUnit.MILLIS) val updatedAt: Instant? = null,
    @field:Version val version: Long? = 0L
)

**Reply by dstepanov** [reply]
Are there two issues? One bug report with missing deleteAt and one feature/bug that the association is not using the optimistic locking? The best would be submit issues with an example that reproduces the problem.

**Reply by cqzhuhui** [reply]
Hi, I will write a example about it to submit. There is only one issue that the association is not using the optimistic locking :)

**Reply by cqzhuhui** [reply]
Hi @dstepanov, I have upload my demo project. You can see the readme in it.
[crud-cascade.zip](https://github.com/micronaut-projects/micronaut-data/files/10398253/crud-cascade.zip)
